### PR TITLE
Changes to adc.c and adc.h to remove errors.

### PR DESCRIPTION
**Pull request by nikolakosaric**
Initialization of QueueMeasurement and CallCallback should be moved to the headerfile rather than being initialized in the .c file

### CODE REVIEW

File: src/adc.c [hunk 1]
```diff
@@ -19,25 +19,34 @@ static struct adc_channel * pending_measurements[ADC_QUEUE_SIZE];
...
+		channels[num_channels].channel = channel;
 		channels[num_channels].callback = callback;
 		channels[num_channels].ptr = ptr;
 		Task_Schedule(QueueMeasurement, &channels[num_channels], period, period);
 		num_channels++;
 	}
 }
 
-static void QueueMeasurement(struct adc_channel * channel) {
+void QueueMeasurement(struct adc_channel * channel) {
```
Comment (gliderman) [inline src/adc.c]: Why did you remove `static` here?  Also applies to the function definition above.

(This comment is to test out review comments)

File: src/adc.c [hunk 2]
```diff
@@ -70,6 +79,6 @@ static void CallCallback(struct adc_channel * channel_struct) {
 	if(channel_struct->ptr) {
 		channel_struct->callback(channel_struct->value, channel_struct->ptr);
 	}else {
-		channel_struct->callback(channel_struct->value);
+		channel_struct->callback(channel_struct->value, 0);
```
Comment (muhlbaier) [inline src/adc.c]: Change this to cast the function pointer to the desired type instead of passing the null pointer to a function that presumably isn't expecting it:
(void(*)(uint16_t)channel_struct->callback(channel_struct->value);

**Comment by muhlbaier** [comment]
Please add static back in. Use a cast instead of passing the 0 into the function that doesn't expect a second pointer input. Also, I don't think task and hal_general need to be included in adc.h.

**Review by nikolakosaric — COMMENTED** [review]
Changed QueueMeasurement back to a static void and fixed the callCallback to cast the function pointer to the desired type rather than passing the pointer to a function that isn't expecting it

**Review by muhlbaier — APPROVED** [review]
Looks good
